pkg/ml/ml.go

```diff
@@ -784,25 +784,12 @@ func NewTensor(ctx *Context, dt DType, dims uint32, ne0, ne1, ne2, ne3 uint32, d
 	return &result
 }
 
-// ggml_permute
+// Permute permutes a tensor, e.g., 1,2,3,4 -> 2,3,4,1
 func Permute(ctx *Context, a *Tensor, axis0, axis1, axis2, axis3 uint32) *Tensor {
 
-	////ASSERT(axis0 >= 0 && axis0 < MAX_DIMS);
```

Comment:
And all the ASSERTs should be moved into special place to re-enable them on the graph init to be sure there no problems with dimensions and data types between tensors.

So yeah, the code is ugly right now, but please do not remove important parts which I'm going to move from there a bit later.